File: News_Service_JDBC/src/main/java/com/dunice/projectNews/services/impl/AuthServiceImpl.java
package com.dunice.projectNews.services.impl;

import com.dunice.projectNews.dto.request.AuthRequest;
import com.dunice.projectNews.dto.request.RegisterUserRequest;
import com.dunice.projectNews.dto.response.LoginUserResponse;
import com.dunice.projectNews.exception.CustomException;
import com.dunice.projectNews.mapper.UserMapper;
import com.dunice.projectNews.models.User;
import com.dunice.projectNews.repository.UserRepositoryJDBC;
import com.dunice.projectNews.security.JwtUserService;
import com.dunice.projectNews.services.AuthService;
import com.dunice.projectNews.util.ServerErrorCodes;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthServiceImpl implements AuthService {

    private final UserRepositoryJDBC userRepository;

    private final UserMapper userMapper;

    private final JwtUserService jwtUserService;

    private final PasswordEncoder passwordEncoder;

    @Override
    public LoginUserResponse registration(RegisterUserRequest registerUserDtoRequest) {
        userRepository.checkEmail(registerUserDtoRequest.getEmail());
        User user = userMapper.registerUserRequestToUser(registerUserDtoRequest);
        user.setPassword(passwordEncoder.encode(user.getPassword()));
        userRepository.saveUser(user);
        String token = jwtUserService.generateToken(user.getEmail());
        LoginUserResponse loginUserDtoResponse = userMapper.userToLoginUserResponse(user);
        loginUserDtoResponse.setToken(token);
        return loginUserDtoResponse;
    }

    @Override
    public LoginUserResponse login(AuthRequest authRequest) {
        User user = userRepository
                .findByEmail(authRequest.getEmail())
                .orElseThrow(() -> new CustomException(ServerErrorCodes.USER_NOT_FOUND));
        if (!passwordEncoder
                .matches(authRequest.getPassword(), user.getPassword())) {
            throw new CustomException(ServerErrorCodes.USER_PASSWORD_NOT_VALID);
        }
        String token = jwtUserService.generateToken(authRequest.getEmail());
        LoginUserResponse loginUserDtoResponse = userMapper.userToLoginUserResponse(user);
        loginUserDtoResponse.setToken(token);
        return loginUserDtoResponse;
    }

}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/services/impl/AuthServiceImpl.kt
package com.dunice.advancedServiceKotlin.services.impl

import com.dunice.advancedServiceKotlin.dto.request.AuthRequest
import com.dunice.advancedServiceKotlin.dto.request.RegisterUserRequest
import com.dunice.advancedServiceKotlin.dto.response.LoginUserResponse
import com.dunice.advancedServiceKotlin.exception.CustomException
import com.dunice.advancedServiceKotlin.mapper.UserMapper
import com.dunice.advancedServiceKotlin.repository.UserRepository
import com.dunice.advancedServiceKotlin.security.JwtUserService
import com.dunice.advancedServiceKotlin.services.AuthService
import com.dunice.advancedServiceKotlin.util.ServerErrorCodes
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class AuthServiceImpl @Autowired constructor(
    private val userRepository: UserRepository,
    private val userMapper: UserMapper,
    private val jwtUserService: JwtUserService,
    private val passwordEncoder: PasswordEncoder
) : AuthService {
    override fun registration(registerUserDtoRequest: RegisterUserRequest): LoginUserResponse {
        if (userRepository.existsByEmail(registerUserDtoRequest.email)) {
            throw CustomException(ServerErrorCodes.USER_ALREADY_EXISTS)
        }
        val user = userMapper.registerUserRequestToUser(registerUserDtoRequest)
        user.password = passwordEncoder.encode(user.password)
        userRepository.save(user)
        val token = jwtUserService.generateToken(user.email)
        val loginUserDtoResponse = userMapper.userToLoginUserResponse(user)
        loginUserDtoResponse.token = token
        return loginUserDtoResponse
    }

    override fun login(authRequest: AuthRequest): LoginUserResponse {
        val user = userRepository.findByEmail(authRequest.email)
            ?: throw CustomException(ServerErrorCodes.USER_NOT_FOUND)
        if (!passwordEncoder
                .matches(authRequest.password, user.password)
        ) {
            throw CustomException(ServerErrorCodes.USER_PASSWORD_NOT_VALID)
        }
        val token = jwtUserService.generateToken(authRequest.email)
        val loginUserDtoResponse = userMapper.userToLoginUserResponse(user)
        loginUserDtoResponse.token = token
        return loginUserDtoResponse
    }
}
